ccloader/js/mod.js

```diff
@@ -10,8 +10,8 @@ export class Mod {
 	 * @param {import('./ccloader').ModLoader} modloader
 	 * @param {string} file 
 	 */
-	constructor(modloader, file){
-		this.file = file;
+	constructor(modloader, baseDirectory){
```

Comment:
Why only provide the directory when the exact file is known? The baseDirectory can always be inferred from the file.

Reply:
Because there is no way to guess which manifest a packed mod contains except try and fetch it. Fetching is done in this class, so I decided to use base directory as a value for identifying separate `Mod` instances. Also, this doesn't have a big impact on performance, yet since I removed `this._getBaseName` this does reduce the amount of code executed.

Reply:
What if we add a new custom command to `serviceworker.js` that acts as a general "Load Manifest + Get Manifest Type"?

Reply:
@ac2pic can you do this?

Reply:
I could, but it won't be with serviceworkers.